File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        prevs = set()
        left = 0
        ans = 0
        for i in range(len(s)):
            while s[i] in prevs:
                prevs.remove(s[left])
                left += 1
            prevs.add(s[i])
            ans = max(i-left+1,ans)
        return ans 
            

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function(s) {
    const freq = Array(26).fill(false);
    let ans = 0
    let l = 0

    for(let i = 0; i < s.length; i++){
        let curPos = s.charCodeAt(i) - 97
        while(freq[curPos]){
            freq[s.charCodeAt(l) - 97] = false
            l += 1
        }
        freq[curPos] = true
        ans = Math.max(ans, i - l + 1);
    }
    return ans
};